controllers/llamastackdistribution_controller.go

```diff
@@ -152,45 +155,81 @@ func (r *LlamaStackDistributionReconciler) SetupWithManager(mgr ctrl.Manager) er
 		Complete(r)
 }
 
-// reconcilePVC creates or updates the PVC for the LlamaStack server.
+// reconcilePVC creates or updates the PersistentVolumeClaim for the LlamaStack server.
+// It generates a dynamic Kustomize overlay in-memory to set the PVC's name and
+// storage size, then uses the generic ApplyKustomizeManifests helper to apply it.
 func (r *LlamaStackDistributionReconciler) reconcilePVC(ctx context.Context, instance *llamav1alpha1.LlamaStackDistribution) error {
-	logger := log.FromContext(ctx)
-
-	// Use default size if none specified
-	size := instance.Spec.Server.Storage.Size
-	if size == nil {
-		size = &llamav1alpha1.DefaultStorageSize
+	if instance.Spec.Server.Storage == nil {
+		return nil
 	}
+	log := log.FromContext(ctx)
 
-	pvc := &corev1.PersistentVolumeClaim{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      instance.Name + "-pvc",
-			Namespace: instance.Namespace,
-		},
-		Spec: corev1.PersistentVolumeClaimSpec{
-			AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
-			Resources: corev1.VolumeResourceRequirements{
-				Requests: corev1.ResourceList{
-					corev1.ResourceStorage: *size,
-				},
-			},
-		},
-	}
+	log.Info("Reconciling PVC")
 
-	if err := ctrl.SetControllerReference(instance, pvc, r.Scheme); err != nil {
-		return fmt.Errorf("failed to set controller reference: %w", err)
-	}
+	// Create a temporary in-memory filesystem for this operation to avoid writing
+	// to the production container's filesystem, which may have restricted permissions.
+	opFs := filesys.MakeFsInMemory()
 
-	found := &corev1.PersistentVolumeClaim{}
-	err := r.Get(ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, found)
+	// Copy the base Kustomize manifests from the reconciler's filesystem (r.Fs)
+	// into the temporary filesystem for this operation. This ensures that in production,
+	// the latest files from the on-disk filesystem are used for every reconciliation.
+	baseDir := "manifests/base"
+	kustFile, err := r.Fs.ReadFile(filepath.Join(baseDir, "kustomization.yaml"))
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			logger.Info("Creating PVC", "pvc", pvc.Name)
-			return r.Create(ctx, pvc)
-		}
-		return fmt.Errorf("failed to fetch PVC: %w", err)
+		return fmt.Errorf("failed to read base kustomization.yaml: %w", err)
 	}
-	// PVCs are immutable after creation, so we don't need to update them
+	pvcFile, err := r.Fs.ReadFile(filepath.Join(baseDir, "pvc.yaml"))
+	if err != nil {
+		return fmt.Errorf("failed to read base pvc.yaml: %w", err)
+	}
+
+	opFs.MkdirAll(baseDir)
```

Comment:
Why do we need this, the [function](https://github.com/llamastack/llama-stack-k8s-operator/pull/54/files#diff-d2fef16128effe03ba3c644a0ead9cdbc473e2a47a8266c9c2f20960798325b1L62) takes manifest directory as input